ydb/core/tx/schemeshard/schemeshard__operation_alter_table.cpp

```diff
@@ -721,14 +733,20 @@ TVector<ISubOperation::TPtr> CreateConsistentAlterTable(TOperationId id, const T
         return {CreateAlterTable(id, tx)};
     }
 
-    TVector<ISubOperation::TPtr> result;
-
-    // only for super user use
-    // until correct and safe altering index api is released
-    if (!IsSuperUser(context.UserToken.Get())) {
+    // Admins can alter indexImplTable unconditionally.
+    // Regular users can only alter allowed fields.
+    if (!IsSuperUser(context.UserToken.Get())
+        && (!CheckAllowedFields(alter, {"Name", "PartitionConfig"})
+            || (alter.HasPartitionConfig()
+                && !CheckAllowedFields(alter.GetPartitionConfig(), {"PartitioningPolicy"})
```

Comment:
This condition is hard to read and comprehend.

Reply:
Will try to simplify in a follow-up PR